### PR TITLE
Reorganize CI jobs

### PR DESCRIPTION
**Pull request by davidcassany**
## Why is this PR needed?

Clean up for the CI tests and structure, part of SUSE/avant-garde#1069

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

This commit reorganized CI jobs to be structuredas follows:

 * caasp-jobs/v4/pr: pull requests jobs
 * caasp-jobs/v4/e2e: e2e and update tests for all platforms
 * caasp-jobs/v4/conformance: conformance tests for all platforms

Fixes SUSE/avant-garde#1298

## Anything else a reviewer needs to know?

Jobs execution history will be lost after the change

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/jenkins/jobs.yaml [hunk 1]
```diff
@@ -15,25 +15,18 @@
...
-      repo-name: skuba
-      repo-owner: SUSE
-      repo-credentials: github-token
-      platform: vmware
+      platform:
+        - openstack
+        - vmware
       jobs:
-          - '{name}-conformance'
+          - '{name}-{platform}-conformance'
```
Comment (pablochacin) [inline ci/jenkins/jobs.yaml]: You have to modify the template for conformance tests and make name match `{name}-{platform}-{conformance}

https://github.com/SUSE/skuba/blob/6b2bb427946eeeae97bb7c723e77b29f8a3679f9/ci/jenkins/templates/conformance-template.yaml#L2

Comment (davidcassany) [reply ci/jenkins/jobs.yaml]: right, just saw the failure on jjb check. Thanks for the hint, lets see if now the test pass.

**Comment by pablochacin** [comment]
Do not merge until communicated to the team.

**Comment by jordimassaguerpla** [comment]
@rsalevsky I know you care about the ui interface. Do these changes look good? Thanks

**Comment by aojea** [comment]
love it, versioned API URL style :smile:

**Pull request merged**